tests/conftest.py

```diff
@@ -0,0 +1,104 @@
+                result = self._resource_call(resource, action, resource_payload, options=options, data=data, files=files)
+                is_foreman_task = isinstance(result, dict) and 'action' in result and 'state' in result and 'started_at' in result
+                if is_foreman_task:
+                    result = self.wait_for_task(result, ignore_errors=ignore_task_errors)
+        except Exception as e:
+            msg = 'Error while performing {0} on {1}: {2}'.format(
+                action, resource, to_native(e))
+            self.fail_from_exception(e, msg)
+        if record_change and not ignore_check_mode:
+            # If we were supposed to ignore check_mode we can assume this action was not a changing one.
+            self.set_changed()
+        return result
+
+    def wait_for_task(self, task, ignore_errors=False):
+        duration = self.task_timeout
+        while task['state'] not in ['paused', 'stopped']:
+            duration -= self.task_poll
+            if duration <= 0:
+                self.fail_json(msg="Timeout waiting for Task {0}".format(task['id']))
+            time.sleep(self.task_poll)
+
+            resource_payload = self._resource_prepare_params('foreman_tasks', 'show', {'id': task['id']})
+            task = self._resource_call('foreman_tasks', 'show', resource_payload)
+        if not ignore_errors and task['result'] != 'success':
+            self.fail_json(msg='Task {0}({1}) did not succeed. Task information: {2}'.format(task['action'], task['id'], task['humanized']['errors']))
+        return task
+
+    def ping(self):
+        return self.resource('ping').call('ping')
+
+
+@pytest.fixture(scope="module")
+def foremanapi():
+    return ForemanApi(
+        uri=f'http://quadlet.tanso.example.com:{FOREMAN_PORT}',
+        username='admin',
+        password='changeme',
+        verify_ssl=False,
+    )
+
+@pytest.fixture
+def organization(foremanapi):
+    org = foremanapi.resource_action('organizations', 'create', params={'organization': {'name': 'Test Org'}})
+    yield org
+    foremanapi.resource_action('organizations', 'destroy', params={'id': org['id']})
+
+@pytest.fixture
+def product(organization, foremanapi):
+    prod = foremanapi.resource_action('products', 'create', params={'name': 'Test Product', 'organization_id': organization['id']})
+    yield prod
+    foremanapi.resource_action('products', 'destroy', params={'id': prod['id']})
+
+@pytest.fixture
+def repository(product, organization, foremanapi):
+    # This is broken as Katello currently does not use the right CA when talking to Pulp
+    # https://github.com/Katello/katello/blob/master/app/models/katello/concerns/smart_proxy_extensions.rb#L254-L265
+    repo = foremanapi.resource_action('repositories', 'create', params={'name': 'Test Repo', 'product_id': product['id'], 'content_type': 'yum'})
+    yield repo
+    foremanapi.resource_action('repositories', 'destroy', params={'id': repo['id']})
```

Comment:
This found #42 and then stuck at `SSL_connect returned=1 errno=0 state=error: certificate verify failed (self-signed certificate in certificate chain)` when trying to talk to Pulp because Katello doesn't inject the CA into Faraday.

Reply:
Look at that! Adding some basic tests are already paying dividends.

Reply:
I just don't like that the dividends are coming out of my account.

Reply:
https://github.com/Katello/katello/pull/8832 was the better fix, now done